### ISSUE TITLE
error when finenune at 'Mixed_8b' layer in inception-resnet-v1

### ISSUE THREAD

**Issue by Victoria2333** [davidsandberg/facenet#826]
I tried to finetune at L-2 layer :
ftune_vlist = [v for v in all_vars if not v.name.startswith('Mixed_8b')] in inception-resnet-v1, 
and train on my_classifier.py to get .pkl file, 
but I got the error: invalid argument 2: size '[16x8x8x8]' is invalid for input with 2048 elements at /pytorch/aten/src/TH/THStorage.c:41 , 
And i found that 'Mixed_8b' corresponds 128 tensor, but 'Logits' corresponds 512 tensor. I don't know why the tensor size is different.
Could you please help me with this?Thanks

**Comment by speculaas** [comment]
Hi Victoria, 

Maybe you want to be more specific?
It looks like you are following the steps of this wiki? github.com/davidsandberg/facenet/wiki/Train-a-classifier-on-own-images

python src/classifier.py TRAIN ~/datasets/my_dataset/train/ ~/models/model-20170216-091149.pb ~/models/my_classifier.pkl --batch_size 1000

If you are, I also tried the same thing.
I didn't encounter this problem. 

If you are not doing : Train-a-classifier-on-own-images , maybe at least post your python command?

BR,
Jimmy

**Comment by Victoria2333** [comment]
@speculaas, Hi ,speculaas, I changed the embedding size to 512 and it worked. But the finetuned accuracy is very low. Here is my specific step:
Here is my code:
all_vars = tf.trainable_variables()
var_to_restore = [v for v in all_vars if not v.name.startswith('Mixed_8b')] in inception-renet-v1
saver = tf.train.Saver(var_to_restore)
I run:
python src/train_softmax.py --data_dir /home/han/facenet/train_face_160 --image_size 160 __embedding_size 512 /home/han/facenet/src/pretrained_models/20180408-102900/model-2018408-102900.ckpt-90
I got a ckpt file(for example, 1.ckpt), i only trained for 4 epoch,coz loss is nearly 0 at 1 epoch...
And I freeze the created 1.ckpt file to 1.pb file which is loaded in my_classifier.py to train and test but the training accuracy is 1 while test accuracy is about 0.4...
I don't know why finetuned model is worse than the pretrained-model(model-2018408-102900.ckpt-90)

Second, I still don't know if following my code, if the layers except Mixed_8b can be trained?? and this layer will be droped?? Am I right???

**Comment by speculaas** [comment]
Hi victoria, 
I never tried finetune only Mixed_8b. 
And yes some part of the graph's trainable variables is not used when doing inference.
But I haven't memorized them. 
No time to check yet.

accuracy are low maybe because not enough number of epochs?
and you didn't use david's pretrained model?

BR,
Jimmy

**Comment by Victoria2333** [comment]
@speculaas, Hi, I tried to finetune my dataset on david's pretrained model to get higher accuracy, but the resut turns out to be bad. You can see it in  'python src/train_softmax.py ...-pretrained_model...' code.

I want to, for exmaple, freeze the old weights of L-2 layer and restore the weights of remaining 2 layer and finetune the last 2 layers. I don't mean to drop some variables which could lower the accuracy obviously, so how can i finetune?? I want to test from which layer to finetune is best for accuracy...

**Comment by Victoria2333** [comment]
Hi,speculaas, has you ever tried train with the pretrained model  on the original CASIA-WebFace dataset??? I think maybe there is sonthing wrong with the code...
I run:
python src/train_softmax.py --data_dir /home/han/facenet/CASIA-WebFace_align --image_size 160 __embedding_size 512 /home/han/facenet/src/pretrained_models/20180408-102900/model-2018408-102900.ckpt-90
Theoretically, the accuracy of pretrained model trained 90 epoch should be nearly 0, but when I trained on the pretrained model on david's dataset-CASIA-WebFace, the loss is 1.198

**Comment by Victoria2333** [comment]
@speculaas, Hi , speculaas, I've been working on the influence of finetune for face recognition. Here is my code:
all_vars = tf.trainable_variables()
set_A_vars = [v for v in all_vars if v.name.startswith('InceptionResnetV1/Block8')]
saver_set_A = tf.train.Saver(set_A_vars, max_to_keep=3)
saver_set_A_and_B = tf.train.Saver(all_vars, max_to_keep=3)

Then I do the transfer learning by restoring the variables in set A:
saver_set_A.restore(sess, pretrained_model)

And save trained model completely:
save_variables_and_metagraph(sess, saver_set_A_and_B, summary_writer, model_dir, subdir, epoch)

I want to test if the finetune of higher layers may do good to accuracy, but my accuracy on small dataset is very low , about o.318, so I want to know about your progress about this??? Did you find the progress of finetuning higher layers??
